### PR TITLE
Add-ons preparation

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- all :D

### CODE REVIEW

File: src/Main/Main.php [hunk 1]
```diff
@@ -39,4 +40,17 @@ public function register(): void
...
+	/**
+	 * Reguster all the services and trigger custom action hook used for addons.
+	 *
+	 * @return void
+	 */
+	public function registerServices(): void
+	{
+		parent::registerServices();
+
+		// Filter triggered when main forms plugins is loaded to hook add-ons.
```
Comment (piqusy) [inline src/Main/Main.php:53]:
```suggestion
		// Filter triggered when main form's plugins is loaded to hook add-ons.
```

File: src/Misc/SettingsWpml.php [hunk 2]
```diff
@@ -78,13 +95,13 @@ public function isSettingsGlobalValid(): bool
...
 		}
 
 		return [
-			$this->getIntroOutput(self::SETTINGS_TYPE_KEY),
-			$this->settingMiscDisclamer(),
+			UtilsSettingsOutputHelper::getIntro(self::SETTINGS_TYPE_KEY),
+			UtilsSettingsOutputHelper::getMiscDisclamer(),
 			[
 				'component' => 'intro',
 				'introSubtitle' => \__('In order for Eightshift forms to work correctly with WPML, you must enable translations in the WPML settings.', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Misc/SettingsWpml.php:107]:
```suggestion
				'introSubtitle' => \__('In order for Eightshift Forms to work correctly with WPML, you must enable translations in the WPML settings.', 'eightshift-forms'),
```

File: src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php [hunk 3]
```diff
@@ -74,37 +74,35 @@ public function __construct(
...
 	 */
-	protected function submitAction(array $formDataReference)
+	protected function submitAction(array $formDetails)
 	{
-		$formId = $formDataReference['formId'];
+		$formId = $formDetails[UtilsConfig::FD_FORM_ID];
 
 		// Send application to Airtable.
-		$delimiter = AbstractBaseRoute::DELIMITER;
+		$delimiter = UtilsConfig::DELIMITER;
```
Comment (piqusy) [inline src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php:92]:
```suggestion
```

File: src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php [hunk 4]
```diff
@@ -74,37 +74,35 @@ public function __construct(
...
 
 		$response = $this->airtableClient->postApplication(
-			"{$formDataReference['itemId']}{$delimiter}{$formDataReference['innerId']}",
-			$formDataReference['params'],
-			$formDataReference['files'],
+			$formDetails[UtilsConfig::FD_ITEM_ID] . $delimiter . $formDetails[UtilsConfig::FD_INNER_ID],
```
Comment (piqusy) [inline src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php:95]:
```suggestion
			$formDetails[UtilsConfig::FD_ITEM_ID] . UtilsConfig::DELIMITER . $formDetails[UtilsConfig::FD_INNER_ID],
```

File: src/Rest/Routes/Settings/MigrationRoute.php [hunk 5]
```diff
@@ -240,7 +217,7 @@ private function getMigration2To3Forms(): array
 
 		if (!$theQuery->have_posts()) {
 			\wp_reset_postdata();
-			return $this->getApiErrorOutput(\__('We could not find any forms on your project so no migration necesery.', 'eightshift-forms'));
+			return UtilsApiHelper::getApiErrorPublicOutput(\__('We could not find any forms on your project so no migration necesery.', 'eightshift-forms'));
```
Comment (piqusy) [inline src/Rest/Routes/Settings/MigrationRoute.php:220]:
```suggestion
			return UtilsApiHelper::getApiErrorPublicOutput(\__('We could not find any forms in your project so there is nothing to migrate.', 'eightshift-forms'));
```

File: src/Rest/Routes/SubmitValidateStepRoute.php [hunk 6]
```diff
@@ -174,13 +171,13 @@ protected function submitAction(array $formDataReference)
 		}
 
 		return \rest_ensure_response(
-			$this->getApiSuccessOutput(
+			UtilsApiHelper::getApiSuccessPublicOutput(
 				\esc_html__('Step validation is success, you may continue.', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Rest/Routes/SubmitValidateStepRoute.php:175]:
```suggestion
				\esc_html__('Step validation is successful, you may continue.', 'eightshift-forms'),
```

File: src/Misc/SettingsCloudflare.php [hunk 7]
```diff
@@ -78,13 +74,13 @@ public function isSettingsGlobalValid(): bool
...
-			return $this->getSettingOutputNoActiveFeature();
+		if (!UtilsSettingsHelper::isOptionCheckboxChecked(self::SETTINGS_CLOUDFLARE_USE_KEY, self::SETTINGS_CLOUDFLARE_USE_KEY)) {
+			return UtilsSettingsOutputHelper::getNoActiveFeature();
 		}
 
 		return [
-			$this->getIntroOutput(self::SETTINGS_TYPE_KEY),
-			$this->settingMiscDisclamer(),
+			UtilsSettingsOutputHelper::getIntro(self::SETTINGS_TYPE_KEY),
+			UtilsSettingsOutputHelper::getMiscDisclamer(),
```
Comment (piqusy) [inline src/Misc/SettingsCloudflare.php:83]:
```suggestion
			UtilsSettingsOutputHelper::getMiscDisclaimer(),
```

File: src/Misc/SettingsWpml.php [hunk 8]
```diff
@@ -78,13 +95,13 @@ public function isSettingsGlobalValid(): bool
...
-			return $this->getSettingOutputNoActiveFeature();
+		if (!UtilsSettingsHelper::isOptionCheckboxChecked(self::SETTINGS_WPML_USE_KEY, self::SETTINGS_WPML_USE_KEY)) {
+			return UtilsSettingsOutputHelper::getNoActiveFeature();
 		}
 
 		return [
-			$this->getIntroOutput(self::SETTINGS_TYPE_KEY),
-			$this->settingMiscDisclamer(),
+			UtilsSettingsOutputHelper::getIntro(self::SETTINGS_TYPE_KEY),
+			UtilsSettingsOutputHelper::getMiscDisclamer(),
```
Comment (piqusy) [inline src/Misc/SettingsWpml.php:104]:
```suggestion
			UtilsSettingsOutputHelper::getMiscDisclaimer(),
```

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Nothing to add from what Ramljak said, but I'll leave it on Req. changes just in case.
Haven't checked _all_ the UI strings in detail, though

File: src/Blocks/Blocks.php [hunk 9]
```diff
@@ -68,10 +95,15 @@ public function getCustomCategory(array $categories, WP_Block_Editor_Context $bl
 			$categories,
 			[
 				[
-					'slug' => 'eightshift-forms',
+					'slug' => UtilsConfig::BLOCKS_MAIN_CATEGORY_SLUG,
 					'title' => \esc_html__('Eightshift Forms', 'eightshift-forms'),
 					'icon' => 'admin-settings',
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Blocks/Blocks.php:100]: maybe I'm not sure if this function can use custom svg. I will test this in the future, not right now. 

Tnx

File: src/Integrations/Mailer/Mailer.php [hunk 10]
```diff
@@ -68,60 +63,65 @@ public function sendFormEmail(
...
+				'wp' => \get_bloginfo('version'),
+				'url' => \get_bloginfo('url'),
+				'userAgent' => isset($_SERVER['HTTP_USER_AGENT']) ? \sanitize_text_field(\wp_unslash($_SERVER['HTTP_USER_AGENT'])) : '',
+				'time' => \wp_date('Y-m-d H:i:s'),
+			],
 		];
 
 		// translators: %1$s replaces the integration name and %2$s formId.
-		$subject = \sprintf(\__('Failed %1$s integration: %2$s', 'eightshift-forms'), $integration, $formId);
+		$subject = \sprintf(\__('Failed %1$s form: %2$s', 'eightshift-forms'), $type, $formId);
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Integrations/Mailer/Mailer.php:108]: this comes to the user so i will leave it as is

File: src/Integrations/Mailer/Mailer.php [hunk 11]
```diff
@@ -68,60 +63,65 @@ public function sendFormEmail(
...
 		$body = '<p>' . \esc_html__('It seems like there was an issue with the user\'s form submission. Here is all the data for debugging purposes.', 'eightshift-forms') . '</p>';
 
 		if ($isDisabled) {
-			$body = '<p>' . \esc_html__('It appears that your integration is currently inactive, and as a result, all the data from your form is included in this email for you to manually input.', 'eightshift-forms') . '</p>';
+			$body = '<p>' . \esc_html__('It appears that your form is currently inactive, and as a result, all the data from your form is included in this email for you to manually input.', 'eightshift-forms') . '</p>';
 			// translators: %1$s replaces the integration name and %2$s formId.
-			$subject = \sprintf(\__('Disabled %1$s integration: %2$s', 'eightshift-forms'), $integration, $formId);
+			$subject = \sprintf(\__('Disabled %1$s form: %2$s', 'eightshift-forms'), $type, $formId);
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Integrations/Mailer/Mailer.php:114]: the same as before

File: src/Misc/SettingsCloudflare.php (hunk 7 above)
Comment (iruzevic) [reply src/Misc/SettingsCloudflare.php:83]: can't change the function name like that, this needs to be changed on the origin. tnx for the comment I will change it

File: src/Misc/SettingsWpml.php (hunk 8 above)
Comment (iruzevic) [reply src/Misc/SettingsWpml.php]: done

File: src/Misc/SettingsCloudflare.php (hunk 7 above)
Comment (iruzevic) [reply src/Misc/SettingsCloudflare.php]: done

File: src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php (hunk 4 above)
Comment (iruzevic) [reply src/Rest/Routes/Integrations/Airtable/FormSubmitAirtableRoute.php:95]: done

**Review by goranalkovic-infinum — APPROVED** [review]

File: src/Misc/SettingsCloudflare.php (hunk 7 above)
Comment (piqusy) [reply src/Misc/SettingsCloudflare.php]: Ofc, tnx

**Review by piqusy — DISMISSED** [review]

**Pull request merged**